CMakeLists.txt

```diff
@@ -16,6 +16,17 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 option(BUILD_TESTS "Option to build unit and device test programs" ON)
 option(DISABLE_DEBUG_PRINTS "Option to disable debug prints, default off" OFF)
 
+#TODO don't want to make all api calls public just for tests...
```

Comment:
Can you work though this TODO to ensure that builds that have BUILD_TESTS enabled don't result in all public methods?

Reply:
You mean to have a getter function to access protected/private info, rather than overloading to be all public?

Reply:
Sometimes for unit testing, it's nice to use parts of the API that are private that should stay private in the API. I was thinking more along the lines of this:

* if BUILD_TESTS, build a version of the library with `PROTECTED=public` and `PRIVATE=public` called something like `jsd-test` 
* Then `test/CMakeLists.txt` would link against `jsd-test` rather than the default `jsd` library, availing the GTests access to private and protected fields and methods. 
* Regardless of `BUILD_TESTS`, always build `jsd` lib with defined `PROTECTED=protected` and `PRIVATE=private` for external consumption. 

Does that make sense?